### PR TITLE
Feat ssr showcases: next, nuxt, angular-ssr

### PR DESCRIPTION
**Pull request by bruno-sch**
## Proposed changes

Adds SSR showcases: next, nuxt, angular ssr

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: scripts/dev.js [hunk 1]
```diff
@@ -34,15 +43,51 @@ inquirer
...
+
+		const currentAnswers = isPlainHtmlSelected
+			? answers.frameworks.filter((a) => a !== 'plain-html')
+			: answers.frameworks;
+
+		if (isPlainHtmlSelected) {
+			startCommand += ' dev:plain-html';
+		}
+
+		const answersFrameworkPairs = [
```
Comment (nmerget) [inline scripts/dev.js:62]: You can move this outside of the function

File: scripts/dev.js [hunk 2]
```diff
@@ -34,15 +43,51 @@ inquirer
...
+				startCommand += ` dev:${framework}-components`;
+			}
+		}
+
+		for (const currentAnswer of currentAnswers) {
+			startCommand += ` start-showcase:${currentAnswer}`;
+		}
 
-		// TODO: Handle child process better
+		// To-do: Handle child process better
```
Comment (nmerget) [inline scripts/dev.js:91]: Why did you change this?

File: scripts/dev.js [hunk 3]
```diff
@@ -34,15 +43,51 @@ inquirer
...
+
 		let startCommand = 'npm-run-all -p start:foundations dev:sass';
-		if (answers?.frameworks)
-			for (const answer of answers.frameworks) {
-				startCommand +=
-					answer === 'plain-html'
-						? ` dev:plain-html`
-						: ` dev:${answer}-components start-showcase:${answer}`;
+
+		const isPlainHtmlSelected = answers.frameworks.includes('plain-html');
```
Comment (nmerget) [inline scripts/dev.js:52]: We could add `start-showcase:plain-html` to  [scripts/package.json](https://github.com/db-ui/mono/pull/2643/files#diff-d74f97384f03053c517e34bcd695eefd751944cd55183d693ae99ef3bc50d016) so we could skip this part, right?

File: scripts/package.json [hunk 4]
```diff
@@ -43,6 +46,9 @@
     "rm-builds:vanilla-public": "rimraf -g showcases/vanilla-showcase/public",
     "rm:builds": "npm-run-all -p rm-builds:*",
     "start-showcase:angular": "npm run dev --workspace=angular-showcase",
+    "start-showcase:angular-ssr": "npm run dev --workspace=angular-ssr-showcase",
```
Comment (nmerget) [inline scripts/package.json:49]: This is starting the showcase as SPA, which isn't different from the normal angular showcase. We need to start the server here somehow

File: showcases/nuxt-showcase/package.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (nmerget) [inline showcases/nuxt-showcase/package.json:1]: Can we disable the telemetry stuff as a default?

`Nuxt collects completely anonymous data about usage.                                                                                                                     09:46:44  
  This will help us improve Nuxt developer experience over time.
  Read more on https://github.com/nuxt/telemetry


> Are you interested in participating?
> Yes /   No
`

**Comment by nmerget** [comment]
Inside `patternhub` we are redireted for every `overview` page to the url with additional params`?density=regular&color=neutral-bg-lvl-1`. I guess this should only happen for the `react-showcase`

File: showcases/patternhub/package.json [hunk 6]
```diff
@@ -9,12 +9,11 @@
...
     "compile": "npm-run-all compile:*",
     "compile:01_jsx": "node scripts/generate-example-jsx.js",
-    "compile:02_copy-components": "cpr ../../output/react/src ./components/src -o",
+    "compile:02_copy-utils": "cpr ../react-showcase/src/utils/base-path.ts ./utils/ -o",
     "compile:03_copy-docs": "cpr ../../docs ./public/docs -o",
     "compile:04_components": "node scripts/esbuild-generate.mjs",
     "compile:05_run": "node generated.js",
-    "compile:06_replace_base": "node scripts/fix-base-path.js",
-    "dev": "cross-env NEXT_PUBLIC_BASE_PATH=/mono/sub npm-run-all compile -p open next:dev",
+    "dev": "cross-env NEXT_PUBLIC_BASE_PATH=/mono/sub NODE_OPTIONS='--inspect' npm-run-all compile:* -p open next:dev",
```
[the remark this answers is not in the document]
Comment (bruno-sch) [reply showcases/patternhub/package.json:15]: This enables us to debug / log next on the server side as well. To be able to fix SSR errors more easily, for example.

File: showcases/next-showcase/.nvmrc [hunk 7]
```diff
@@ -0,0 +1 @@
+18
```
[the remark this answers is not in the document]
Comment (bruno-sch) [reply showcases/next-showcase/.nvmrc:1]: Nope, not really. Rather counterproductive...

File: scripts/dev.js (hunk 3 above)
Comment (bruno-sch) [reply scripts/dev.js:52]: i have simplified the function. please check whether it works for you.

File: scripts/package.json (hunk 4 above)
Comment (bruno-sch) [reply scripts/package.json:49]: To me it looks like Angular is started in ssr mode with this command because:

- browsers console logs: "Angular hydrated 1 component(s) and 5 node(s), 0 component(s) were skipped. ..."
- terminal logs errors such as "ERROR ReferenceError: document is not defined", which should only happen on the server side

File: showcases/nuxt-showcase/package.json (hunk 5 above)
Comment (bruno-sch) [reply showcases/nuxt-showcase/package.json:1]: I added `telemetry: false` to nuxt.config.ts

File: scripts/package.json (hunk 4 above)
Comment (nmerget) [reply scripts/package.json:49]: You are right, but why is the 'id'-problem working in angular :O

File: showcases/next-showcase/pages/[[...slug]].tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+				}
+			}
+
+			if (component) {
+				accumulator.push({ path, component });
+			}
+
+			return accumulator;
+		},
+		[]
```
Comment (nmerget) [inline showcases/next-showcase/pages/[[...slug]].tsx:87]: This is the same as `getStaticPaths` isn't it?

File: showcases/react-showcase/src/meta-navigation/index.tsx [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import { useEffect, useState } from 'react';
-import { useSearchParams } from 'react-router-dom';
+// Import { useSearchParams } from 'react-router-dom';
```
Comment (nmerget) [inline showcases/react-showcase/src/meta-navigation/index.tsx:2]: We can remove this right?

File: showcases/shared/default-component-data.ts [hunk 10]
```diff
@@ -1,3 +1,5 @@
+import { type FunctionComponent, type ReactElement } from 'react';
```
Comment (nmerget) [inline showcases/shared/default-component-data.ts:1]: This file is shared between all showcases even for angular and vue. We shouldn't use a react dependency here

File: scripts/dev.js [hunk 11]
```diff
@@ -16,11 +35,20 @@ inquirer
 				{
 					name: 'angular'
 				},
+				{
+					name: 'angular-ssr'
+				},
 				{
```
Comment (nmerget) [inline scripts/dev.js:41]: Maybe we can add patternhub here as well?

File: scripts/package.json (hunk 4 above)
Comment (bruno-sch) [reply scripts/package.json:55]: Perhaps we can be optimistic and angular ssr simply works in this respect?

File: showcases/next-showcase/pages/[[...slug]].tsx (hunk 8 above)
Comment (bruno-sch) [reply showcases/next-showcase/pages/[[...slug]].tsx]: Yes, almost. The loops have the same structure, but the return values are structured differently...

I have outsourced the loop to a function. This has the parameter `accumulateFunction` (using TS generics), so it can return different return values using an (preferably) anonymous arrow-function as parameter.

File: showcases/shared/default-component-data.ts (hunk 10 above)
Comment (bruno-sch) [reply showcases/shared/default-component-data.ts]: I have created the type ReactDefaultComponentProps for the react-showcases (react, next, patternhub), which extends / overwrites DefaultComponentProps.

File: scripts/dev.js (hunk 11 above)
Comment (bruno-sch) [reply scripts/dev.js:41]: Yes, patternhub is now also available here.

File: scripts/package.json [hunk 12]
```diff
@@ -49,7 +52,11 @@
...
+    "start-showcase:next": "npm run dev --workspace=next-showcase",
+    "start-showcase:nuxt": "npm run dev --workspace=nuxt-showcase",
     "start-showcase:patternhub": "npm run dev --workspace=patternhub",
+    "start-showcase:plain-html": "",
```
Comment (nmerget) [inline scripts/package.json:59]: This script is empty

File: showcases/patternhub/scripts/generate-docs-mdx.js [hunk 13]
```diff
@@ -109,12 +89,12 @@ const generateDocsMdx = async () => {
 				);
 			}
 
-			if (!FS.existsSync(`${componentOldPath}/index.tsx`)) {
-				FS.writeFileSync(
-					`${componentOldPath}/index.tsx`,
-					getRedirectOldFiles('../../../')
-				);
-			}
+			// If (!FS.existsSync(`${componentOldPath}/index.tsx`)) {
```
Comment (nmerget) [inline showcases/patternhub/scripts/generate-docs-mdx.js:92]: We can remove this right?

File: showcases/patternhub/scripts/generate-docs-mdx.js [hunk 14]
```diff
@@ -123,12 +103,11 @@ const generateDocsMdx = async () => {
 				);
 			}
 
-			if (!FS.existsSync(`${componentOldPath}/overview.tsx`)) {
-				FS.writeFileSync(
-					`${componentOldPath}/overview.tsx`,
-					getRedirectOldFiles('../../../')
-				);
-			}
+			// If (!FS.existsSync(`${componentOldPath}/overview.tsx`)) {
```
Comment (nmerget) [inline showcases/patternhub/scripts/generate-docs-mdx.js:106]: https://github.com/db-ui/mono/pull/2643/files#r1642551876

File: showcases/patternhub/components/card-navigation/card-navigation.tsx [hunk 15]
```diff
@@ -2,14 +2,15 @@ import { useRouter } from 'next/router';
...
-import { DBCard, DBIcon } from '../src';
+import { DBCard, DBIcon } from '../../../../output/react/src';
 
 const CardNavigation = () => {
 	const router = useRouter();
 
 	const navigationItemParent = getAllNavigationItems().find(
 		(navItem) =>
-			navItem.path === router.pathname.replace('/[[...slug]]', '')
+			// NavItem.path === router.pathname.replace('/[[...slug]]', '')
```
Comment (nmerget) [inline showcases/patternhub/components/card-navigation/card-navigation.tsx:12]: Let's remove this comment

File: scripts/package.json (hunk 12 above)
Comment (bruno-sch) [reply scripts/package.json]: Fixed :)

File: showcases/patternhub/scripts/generate-docs-mdx.js (hunk 14 above)
Comment (bruno-sch) [reply showcases/patternhub/scripts/generate-docs-mdx.js:106]: @nmerget I don't understand what you're trying to tell me here... Can you please explain your thoughts?

File: showcases/patternhub/components/card-navigation/card-navigation.tsx (hunk 15 above)
Comment (bruno-sch) [reply showcases/patternhub/components/card-navigation/card-navigation.tsx]: Yes :)

**Review by nmerget — APPROVED** [review]
